djamazing/storage.py

```diff
@@ -29,35 +31,74 @@ def check_signature(signature, filename, username):
     return True
 
 
+class S3File(object):
+    def __init__(self, name, s3_object):
+        self.name = name
+        self.data = s3_object.get()
+
+    @property
+    def size(self):
+        return self.data['ContentLength']
+
+    @property
+    def file(self):
+        return self
+
+    @property
+    def open(self):
+        return self
+
+    def read(self, num_bytes=None):
+        return self.data['Body'].read(num_bytes)
+
+    def chunks(self, chunk_size=None):
+        while True:
+            chunk = self.read(chunk_size)
+            if chunk:
+                yield chunk
+            else:
+                return
+
+    def __iter__(self):
+        for chunk in self.chunks():
+            for line in chunk.split('\n'):
+                yield line
+
+
```

Comment:
In general looks good for simple readonly things.

I'm wondering if there would be any usecase to have `write` method in the future like here:

https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto.py#L69